### DISCUSSION TITLE
Have custom field in Site add up values of cost assigned to each circuit

### DISCUSSION THREAD

**Discussion by ben-marquardt-tfs** [netbox-community/netbox#9419] [Q&A] [answered]
Hi, I am new to Netbox and am trying to figure out if it is possible to have a custom field of a cost for each circuit and then have another custom field at the site level that adds up the costs of all the circuits associated with this site. 

For example, if there are two circuits:
Circuit A has a cost of 10
Circuit B has a cost of 20

Both are in Site 1

Is it possible to have custom fields that automatically add up the cost so the custom field on Site 1 has a cost of 30?

Thanks for any help!

**Comment by mtinberg** [comment, marked as answer]
It's not possible for custom fields to work this way without writing additional code, you could create a Netbox Report which collects all the per-circuit custom field data and shows the Site totals though, or use an external script over the API to both calculate the totals you want, but also populate a custom field on Site with the calculated data.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: benmarquardtTFS ***@***.***>
Sent: Tuesday, May 24, 2022 1:46 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Have custom field in Site add up values of cost assigned to each circuit (Discussion #9419)


Hi, I am new to Netbox and am trying to figure out if it is possible to have a custom field of a cost for each circuit and then have another custom field at the site level that adds up the costs of all the circuits associated with this site.

For example, if there are two circuits:
Circuit A has a cost of 10
Circuit B has a cost of 20

Both are in Site 1

Is it possible to have custom fields that automatically add up the cost so the custom field on Site 1 has a cost of 30?

Thanks for any help!

**Reply by ben-marquardt-tfs** [reply]
Thanks! This may be a tough question for you to answer, but as someone who is new to Netbox it is hard for me to figure out the easiest way to go about this. Would you say writing an external script over the API or using a Netbox Report would be the best solution? I can of course try both of them but I am wondering where you would recommend I start.

**Reply by BuddyDoDat** [reply]
Ben, how did this work out for you in the end?  I'm looking to do the exact same thing.

**Comment by mtinberg** [comment]
I don't have a recommendation on that front, both are useful skills that you will probably end up wanting to use, so if you think you have other near-term goals to write Reports then learn that first, or if you want to do bulk programmatic changes, import/export, interface with other systems, then learning the REST API first makes sense.  Either way the skill will pay off.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
